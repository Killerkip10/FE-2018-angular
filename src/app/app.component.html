<app-header></app-header>

<mat-drawer-container class="container">
  <mat-drawer *ngIf="tokenService.getToken()" mode="side" opened>
    <mat-nav-list>
      <a mat-list-item [routerLink]="['user-profile']" routerLinkActive="active">Profile</a>
      <a mat-list-item [routerLink]="['user-list']" routerLinkActive="active">Users list</a>

    </mat-nav-list>
  </mat-drawer>

  <mat-drawer-content>
    <router-outlet></router-outlet>
  </mat-drawer-content>
</mat-drawer-container>

<app-footer></app-footer>
