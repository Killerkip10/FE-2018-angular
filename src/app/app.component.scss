app-header{
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 2;
}
.container {
  margin-top: 56px;
  min-height: calc(100vh - 112px);
  mat-drawer{
    position: fixed;
    top: 56px;
    bottom: inherit;
  }
  .active {
    background-color: rgba(0,0,0,.04);
  }
}
