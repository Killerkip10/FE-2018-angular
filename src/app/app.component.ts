import {Component} from '@angular/core';
import {TranslateService} from '@ngx-translate/core';

import {TokenService} from './core/services';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  private tokenService = TokenService;

  constructor(translateService: TranslateService) {
    translateService.addLangs(['en', 'ru']);
    translateService.setDefaultLang('en');
  }
}
