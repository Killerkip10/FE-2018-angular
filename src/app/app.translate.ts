import {NgModule} from '@angular/core';
import {TranslateModule, TranslateLoader} from '@ngx-translate/core';
import {TranslateHttpLoader} from '@ngx-translate/http-loader';
import {HttpClient} from '@angular/common/http';

@NgModule({
  imports: [
    TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: HttpLoaderFactory,
        deps: [HttpClient]
      }
    })
  ]
})
export class AppTranslate {}

function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(http, 'http://localhost:3030/i18n/', '.json');
}
