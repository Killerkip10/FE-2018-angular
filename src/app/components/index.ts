export * from './user-edit-form/user-edit-form.module';
export * from './user-info/user-info.module';
export * from './login-form/login-form.module';
export * from './forgot-password-form/forgot-password-form.module';
