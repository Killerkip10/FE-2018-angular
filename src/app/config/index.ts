export * from './url-config';
export * from './http-config';
