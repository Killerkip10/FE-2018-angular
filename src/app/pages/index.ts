export * from './user-profile/user-profile-page.module';
export * from './login/login-page.module';
export * from './forgot-password/forgot-password-page.module';
export * from './user-list/user-list.module';
