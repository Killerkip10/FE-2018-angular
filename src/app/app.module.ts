import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {RouterModule} from '@angular/router';
import {TranslateModule} from '@ngx-translate/core';

import {CoreModule} from './core/core.module';
import {AppRouting} from './app.routing';
import {AppComponent} from './app.component';
import {AppTranslate} from './app.translate';
import {
  MatSidenavModule,
  MatListModule
} from '@angular/material';
import {
  LoginPageModule,
  UserProfilePageModule,
  ForgotPasswordPageModule,
  SearchUsersPageModule
} from './pages';


@NgModule({
  imports: [
    CoreModule,
    CommonModule,
    RouterModule,
    TranslateModule,

    LoginPageModule,
    UserProfilePageModule,
    ForgotPasswordPageModule,
    SearchUsersPageModule,

    MatSidenavModule,
    MatListModule,

    AppTranslate,
    AppRouting
  ],
  declarations: [
    AppComponent
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
