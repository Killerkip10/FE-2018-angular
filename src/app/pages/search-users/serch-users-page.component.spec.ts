import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {CommonModule} from '@angular/common';

import {SearchUsersPageComponent} from './search-users-page.component';
import {AppTranslate} from '../../app.translate';

describe('SearchUsersPageComponent', () => {
  let component: SearchUsersPageComponent;
  let fixture: ComponentFixture<SearchUsersPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        CommonModule,
        HttpClientModule,
        TranslateModule,

        AppTranslate
      ],
      declarations: [
        SearchUsersPageComponent
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SearchUsersPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.tabs = [];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.tabs = ['TAB1', 'TAB2', 'TAB3', 'TAB4'];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.tabIndex = 0;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.tabIndex = 1;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
