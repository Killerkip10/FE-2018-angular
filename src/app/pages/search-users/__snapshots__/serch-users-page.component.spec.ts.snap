// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchUsersPageComponent snapshot 1`] = `
<app-search-user-page
  tabIndex="0"
  tabs={[Function Array]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    <div
      class="container"
      ng-reflect-ng-switch="0"
    >
      
      <app-user-list />
      
    </div>
  </section>
</app-search-user-page>
`;

exports[`SearchUsersPageComponent snapshot 2`] = `
<app-search-user-page
  tabIndex="0"
  tabs={[Function Array]}
>
  <section>
    <mat-tab-group>
      
    </mat-tab-group>
    <div
      class="container"
      ng-reflect-ng-switch="0"
    >
      
      <app-user-list />
      
    </div>
  </section>
</app-search-user-page>
`;

exports[`SearchUsersPageComponent snapshot 3`] = `
<app-search-user-page
  tabIndex="0"
  tabs={[Function Array]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    <div
      class="container"
      ng-reflect-ng-switch="0"
    >
      
      <app-user-list />
      
    </div>
  </section>
</app-search-user-page>
`;

exports[`SearchUsersPageComponent snapshot 4`] = `
<app-search-user-page
  tabIndex="0"
  tabs={[Function Array]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    <div
      class="container"
      ng-reflect-ng-switch="0"
    >
      
      <app-user-list />
      
    </div>
  </section>
</app-search-user-page>
`;

exports[`SearchUsersPageComponent snapshot 5`] = `
<app-search-user-page
  tabIndex={[Function Number]}
  tabs={[Function Array]}
>
  <section>
    <mat-tab-group>
      
      <mat-tab />
      <mat-tab />
    </mat-tab-group>
    <div
      class="container"
      ng-reflect-ng-switch="1"
    >
      
      
      <app-item-list />
    </div>
  </section>
</app-search-user-page>
`;
